src/components/TodoContext.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable */
+import React, {
+  Dispatch,
+  useEffect,
+  useReducer,
+} from "react";
+import { Todo } from "../types/Todo";
+import { getTodos } from '../api/todos';
+
+const USER_ID = 11384;
+
+export enum ACTIONS {
+  SORT,
+  SET_LENGTH,
+  SET_LIST,
+  SET_ERROR,
+}
+
+export enum FILTER {
+  ALL = 'All',
+  ACTIVE = 'Active',
+  COMPLETED = 'Completed',
+}
+
+type Action = { type: ACTIONS.SORT, payload: string }
+  | { type: ACTIONS.SET_LIST, payload: Todo[] }
+  | { type: ACTIONS.SET_LENGTH, payload: number }
+  | { type: ACTIONS.SET_ERROR, payload: string }
+
+interface Data {
+  list: Todo[],
+  sortBy: string,
+  totalLength: number,
+  error: string,
+};
+
+type State = [
+  state: Data,
+  dispatch: Dispatch<Action>,
+];
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case ACTIONS.SORT:
+      return [
+        {
+          ...state[0],
+          sortBy: action.payload,
+        },
+        () => { }
+      ];
+    case ACTIONS.SET_LIST:
+      return [
+        {
+          ...state[0],
+          list: action.payload,
+        },
+        () => { }
+      ];
+    case ACTIONS.SET_ERROR:
+      return [
+        {
+          ...state[0],
+          error: action.payload,
+        },
+        () => { }
+      ];
+    case ACTIONS.SET_LENGTH:
+      return [
+        {
+          ...state[0],
+          totalLength: action.payload,
+        },
+        () => { }
+      ];
+    default:
+      return state;
+  }
+}
+
+const initialState: State = [
+  {
+    list: [],
+    sortBy: 'All',
+    totalLength: 0,
+    error: '',
+  },
+  () => { },
+];
+
+export const StateContext = React.createContext(initialState);
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialState)
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(res => {
+        dispatch({ type: ACTIONS.SET_LIST, payload: res })
+        dispatch({ type: ACTIONS.SET_LENGTH, payload: res.length })
```

Comment:
you need to combine these two dispatches to one

Reply:
can you give a hint how to do it properly ?)